### DISCUSSION TITLE
Support for other remote Git Repositories?

### DISCUSSION THREAD

**Discussion by Andre601** [squidfunk/mkdocs-material#5731] [Q&A] [answered]
Material for MkDocs currently supports GitHub for displaying stats such as stars, forks and latest release on the site... My question is, if this can be configured and/or extended to support other remote Git hosts such as GitLab, BitBucket, Codeberg, etc.

Preferrably, support for any Gitea/Forgejo instance through matching configuration would be wonderful.

If for whatever reason it isn't doable on your end directly, is there a way to achieve this using theme extension alone? Or would a plugin be necessary to achieve this?

Thanks in advance for a reply.

**Comment by squidfunk** [comment, marked as answer]
Yes, this can be extended! We're already offering support for GitHub and GitLab, which are the most popular coding platforms. We're currently not planning to add support for other, lesser known platforms, because there are just not as many users that would justify the increased maintenance burden, but you can certainly fork the theme and add it on your fork!

GitHub is here:

https://github.com/squidfunk/mkdocs-material/blob/c2e55585fa4ab4657d8c47ca51e83caee550f962/src/assets/javascripts/components/source/facts/github/index.ts#L60-L103

And integration is here:

https://github.com/squidfunk/mkdocs-material/blob/c2e55585fa4ab4657d8c47ca51e83caee550f962/src/assets/javascripts/components/source/facts/_/index.ts#L68-L88

**Reply by Andre601** [reply]
What about a general support for Gitea/Forgejo-based APIs?
They are pretty much the same for the stats you get, so it could be one single system. Only requirement would then be for the user to specifically tell MkDocs Material that the repo uses a gitea-based software and the Domain used...

Tho, can understand that this may be a lot of extra work.

**Comment by Andre601** [comment]
Would you be willing to look at a PR if I implement support?

Tho, I consider to try and implement a general support for any forgejo/gitea instance, so that people can display stats from a repository of their own instance.
The only question here would be what a optimal regex pattern - if any - would be to use here, as with forgejo/gitea any domain pattern would be a possible solution... Only the API path may be consistent...
